### DISCUSSION TITLE
[taglib] unable to use find_package to find the Taglib package

### DISCUSSION THREAD

**Discussion by altschuler** [microsoft/vcpkg#32104] [Q&A] [answered]
I've installed `taglib` with manifest mode, but am unable to use `find_package(Taglib)` (I've tried taglib, Taglib, TagLib, etc). I get the following error

```
CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:853 (_find_package):
  Could not find a package configuration file provided by "Taglib" with any
  of the following names:

    TaglibConfig.cmake
    taglib-config.cmake

  Add the installation prefix of "Taglib" to CMAKE_PREFIX_PATH or set
  "Taglib_DIR" to a directory containing one of the above files.  If "Taglib"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:40 (find_package)
```

There seems to be no "taglib" variables exported at all, `Taglib_DIR` has the value `Taglib_DIR-NOTFOUND`.

I've had that issue before with other packages, but some combination of rebuilding/reinstalling, executing the bootstrap script or random other fiddling solved it so far. I see a `taglib-config.cmake` in `./vcpkg/buildtrees/taglib/src/v1.13-5298fc2c7f.clean/taglib-config.cmake`, and another in `./vcpkg/buildtrees/taglib/x64-linux-rel/taglib-config` (not a .cmake file).

Am I missing something? Thanks!

**Comment by dg0yt** [comment, marked as answer]
vcpkg ports do not generally offer CMake packages. Normally it is a responsibility of the upstream developers. There are also some standard Find modules in CMake, and for some ports, vcpkg maintainers added unofficial Cmake config packages.

Taglib comes with official pkg-config files. It can be imported via the CMake `FindPkgConfig` module (untested):
~~~cmake
    find_package(PkgConfig)
    # For taglib
    pkg_check_modules(TAGLIB  REQUIRED IMPORTED_TARGET taglib)
    target_link_libraries(main PRIVATE PkgConfig::TAGLIB)
    # For C bindings
    pkg_check_modules(TAGLIB_C REQUIRED IMPORTED_TARGET taglib_c)
    target_link_libraries(main PRIVATE PkgConfig::TAGLIB_C)
~~~~
For this to work, you must have pkg-config or pkgconf installed and available to your cmake project.

**Reply by altschuler** [reply]
Thank you very much! It works perfectly. I've opened a [PR](https://github.com/microsoft/vcpkg/pull/32128) to add this to the usage file for the taglib port.
